File: de.genesez.example.php/src-gen/base/Car.php
<?php
/**
 * @internal <!-- PROTECTED REGION ID(php.own.file.comment._11_5_6340215_1177945913718_872802_139) ENABLED START -->
 * @internal <!-- TODO: put your own comments here -->
 * @internal <!-- PROTECTED REGION END -->
 * @author	dreamer
 * @package	base
 */

require_once '../base/Wheel.php';
require_once '../base/License.php';
require_once '../base/LicensePlate.php';
require_once '../base/Engine.php';
require_once '../base/NotReallySeriousException.php';
require_once '../base/Vehicle.php';

/* PROTECTED REGION ID(php.own.imports._11_5_6340215_1177945913718_872802_139) ENABLED START */
// TODO: put your further include + require statements here
/* PROTECTED REGION END */

/**
 * A car is a vehicle with four wheels and drives on the road.
 * 
 * @internal <!-- PROTECTED REGION ID(php.own.type.comment._11_5_6340215_1177945913718_872802_139) ENABLED START -->
 * @internal <!-- TODO: put your own comments here -->
 * @internal <!-- PROTECTED REGION END -->
 * @author	dreamer
 * @package	base
 */
class Car extends Vehicle  {
	
	// -- attributes --------------------------------------------------------
	/**
	 * @var		int	$maxSpeed	describs the maximum speed of a car
	 * @see		Car::getMaxSpeed()
	 */
	private $maxSpeed;
	
	// -- associations ------------------------------------------------------
	/**
	 * @var		Wheel	$spare	reference to the spared wheel
	 * @see		Car::getSpare()
	 */
	private $spare;
	/**
	 * @var		License	$license	
	 * @see		Car::getLicense()
	 */
	private $license;
	/**
	 * @var		LicensePlate	$licensePlate	
	 * @see		Car::getLicensePlate()
	 */
	private $licensePlate;
	/**
	 * @var		Engine	$engine	
	 * @see		Car::getEngine()
	 */
	private $engine;
	/**
	 * @var		Wheel	$wheels	reference to normal wheels on a car
	 * @see		Car::getWheels()
	 */
	private $wheels = array();
	
	// -- constructors + destructors ----------------------------------------
	/**
	 * constructs an object of class Car
	 * 
	 * constructor of the class car
	 * 
	 * the constructor may throw the following exceptions:
	 * {@link NotReallySeriousException NotReallySeriousException}
	 * @throws		NotReallySeriousException
	 * 
	 * @param	int	$mxs	maximum of speed
	 * @param	string	$make	i don't know what make do
	 */
	public function __construct($mxs, $make) {
		/* PROTECTED REGION ID(php.constructor._11_5_6340215_1179229080562_965427_117) ENABLED START */
		// TODO: implementation of method 'Car.car(...)'
		/* PROTECTED REGION END */
	}
	
	
	// -- method implementations --------------------------------------------
	/**
	 * @internal <!-- PROTECTED REGION ID(php.own.operation.comment._12_5_b6f02e1_1204102049578_778385_691) ENABLED START -->
	 * @internal <!-- TODO: put your own comments here -->
	 * @internal <!-- PROTECTED REGION END -->
	 * @param	int	$distance	
	 * @return	boolean	
	 */
	public function drive($distance) {
		$returnValue = NULL;
		/* PROTECTED REGION ID(php.implementation._12_5_b6f02e1_1204102049578_778385_691) ENABLED START */
		// TODO: implementation of method 'Car.drive(...)'
		/* PROTECTED REGION END */
		return $returnValue;
	}
	
	// -- methods for derived attributes ------------------------------------
	/**
	 * getter for the attribute $noOfWheels
	 * 
	 * count the number of wheels
	 * @return	int	the value of $noOfWheels
	 */
	private function getNoOfWheels() {
		$returnValue = NULL;
		/* PROTECTED REGION ID(php.derived.attribute.implementation._11_5_6340215_1178786244296_257674_121) ENABLED START */
		// TODO: implementation of method 'Car.getNoOfWheels(...)'
		/* PROTECTED REGION END */
		return returnValue;
	}
	
	// -- association accessors ---------------------------------------------
	/**
	 * getter for the association attribute $spare
	 * 
	 * reference to the spared wheel
	 * @return	Wheel	the value of $spare
	 */
	private function getSpare() {
		return $spare;
	}
	/**
	 * setter for the association attribute $spare
	 * @param	Wheel	$spare	the value to set
	 * @see		Car::getSpare()
	 */
	private function setSpare(Wheel $spare) {
		self::$spare = $spare;
	}
	/**
	 * getter for the association attribute $license
	 * 
	 * @internal <!-- PROTECTED REGION ID(php.own.association.comment._12_5_6340215_1182161516984_882840_381) ENABLED START -->
	 * @internal <!-- TODO: put your own comments here -->
	 * @internal <!-- PROTECTED REGION END -->
	 * @return	License	the value of $license
	 */
	private function getLicense() {
		return $license;
	}
	/**
	 * setter for the association attribute $license
	 * @param	License	$license	the value to set
	 * @see		Car::getLicense()
	 */
	private function setLicense(License $license) {
		self::$license = $license;
	}
	/**
	 * getter for the association attribute $licensePlate
	 * 
	 * @internal <!-- PROTECTED REGION ID(php.own.association.comment._12_5_8a7027a_1182165461531_196368_421) ENABLED START -->
	 * @internal <!-- TODO: put your own comments here -->
	 * @internal <!-- PROTECTED REGION END -->
	 * @return	LicensePlate	the value of $licensePlate
	 */
	private function getLicensePlate() {
		return $licensePlate;
	}
	/**
	 * setter for the association attribute $licensePlate
	 * @param	LicensePlate	$licensePlate	the value to set
	 * @see		Car::getLicensePlate()
	 */
	private function setLicensePlate(LicensePlate $licensePlate) {
		self::$licensePlate = $licensePlate;
	}
	/**
	 * getter for the association attribute $engine
	 * 
	 * @internal <!-- PROTECTED REGION ID(php.own.association.comment._12_5_8a7027a_1182165513891_613516_500) ENABLED START -->
	 * @internal <!-- TODO: put your own comments here -->
	 * @internal <!-- PROTECTED REGION END -->
	 * @return	Engine	the value of $engine
	 */
	private function getEngine() {
		return $engine;
	}
	/**
	 * setter for the association attribute $engine
	 * @param	Engine	$engine	the value to set
	 * @see		Car::getEngine()
	 */
	private function setEngine(Engine $engine) {
		self::$engine = $engine;
	}
	/**
	 * getter for the association attribute $wheels
	 * 
	 * reference to normal wheels on a car
	 * @return	array	the value of $wheels as an array of type Wheel
	 */
	private function getWheels() {
		return array_values($wheels);
	}
	/**
	 * adds the given value to the association attribute $wheels
	 * @param	Wheel	$toAdd	the value to add
	 * @see		Car::getWheels()
	 */
	private function insertInWheels(Wheel $toAdd) {
		self::$wheels[] = $toAdd;
	}
	/**
	 * removes the given value from the association attribute $wheels
	 * @param	Wheel	$toRemove	the value to remove
	 * @see		Car::getWheels()
	 */
	private function removeFromWheels(Wheel $toRemove) {
		$entry = array_search($toRemove, $wheels);
		if ($entry !== FALSE) {
			unset(self::$wheels[$entry]);
		}
	}
	
	// -- attribute accessors -----------------------------------------------
	/**
	 * getter for the attribute $maxSpeed
	 * 
	 * describs the maximum speed of a car
	 * @return	int	the value of $maxSpeed
	 */
	private function getMaxSpeed() {
		return $maxSpeed;
	}
	/**
	 * setter for the attribute $maxSpeed
	 * @param	int	$maxSpeed	the value to set
	 * @see		Car::getMaxSpeed()
	 */
	private function setMaxSpeed(int $maxSpeed) {
		self::$maxSpeed = $maxSpeed;
	}
	
	// -- own code implementation -------------------------------------------
	/* PROTECTED REGION ID(php.class.own.code.implementation._11_5_6340215_1177945913718_872802_139) ENABLED START */
	// TODO: put your further code implementations for class 'Car' here
	/* PROTECTED REGION END */
}
?>

File: branches/GeneSEZ_1-0-0/de.genesez.example/src-gen/base/Car.java
package base;


/* <!-- PROTECTED REGION ID(java.mclassifier.other.import.code_11_5_6340215_1177945913718_872802_139) ENABLED START --> */
/* <!-- TODO put your imports here --> */
import java.lang.String;

import java.util.Collections;
import java.util.HashSet;
import java.util.Set;

/* <!-- PROTECTED REGION END --> */
/**
 * A car is a vehicle with four wheels and drives on the road.
 * @author nicher
 */
public class Car extends Vehicle {

    // ////////////////////////////////////////////////////////////////////////
    // generated variable, constant and association declarations
    // ////////////////////////////////////////////////////////////////////////

    /** @see {@link getMaxSpeed} */
    private int maxSpeed;

    /**
     * variable for association to wheels
     * @see {@link getWheels}
     */
    private java.util.Set<Wheel> wheels = new java.util.HashSet<Wheel>();

    /**
     * variable for association to spare
     * @see {@link getSpare}
     */
    private Wheel spare;

    /**
     * variable for association to license
     */
    private java.util.Set<Owner> license = new java.util.HashSet<Owner>();

    /**
     * variable for association to licensePlate
     */
    private LicensePlate licensePlate;

    /**
     * variable for association to engine
     */
    private Engine engine;

    // ////////////////////////////////////////////////////////////////////////
    // generated constructor implementations
    // ////////////////////////////////////////////////////////////////////////

    /**
     * constructor of the class car
     * @param  mxs  maximum of speed
     * @param  make  i don't know what make do
     * @throws  NotReallySeriousException  this is a comment; this is the second comment
     */
    public Car(int mxs, String make) throws NotReallySeriousException {

        /* <!-- PROTECTED REGION ID(java.moperation.implementation.operation.code._11_5_6340215_1179229080562_965427_117) ENABLED START --> */
        /* <!-- TODO put your own implementation code here --> */

        /* <!-- PROTECTED REGION END --> */
    }

    // ////////////////////////////////////////////////////////////////////////
    // generated derived attribute method implementations
    // ////////////////////////////////////////////////////////////////////////

    /**
      * count the number of wheels
      */
    private Integer getNoOfWheels() {
        Integer returnValue = null;

        /* <!-- PROTECTED REGION ID(java.mattribute.implementation.attribute.code._11_5_6340215_1178786244296_257674_121) ENABLED START --> */
        /* <!-- TODO put your own implementation code here --> */

        /* <!-- PROTECTED REGION END --> */
        return returnValue;
    }

    // ////////////////////////////////////////////////////////////////////////
    // generated association method implementations
    // ////////////////////////////////////////////////////////////////////////

    /**
     * accessor for association to wheels
     * reference to normal wheels on a car
     */
    public java.util.Set<Wheel> getWheels() {
        return java.util.Collections.unmodifiableSet(wheels);
    }

    /**
     * accessor for association to wheels
     * @see {@link getWheels}
     */
    public void insertInWheels(Wheel _in) {
        if (wheels.contains(_in)) {
            return;
        }
        wheels.add(_in);
        _in.insertInMycar(this);
    }

    /**
     * accessor for association to wheels
     * @see {@link getWheels}
     */
    public void removeFromWheels(Wheel _ex) {
        if (!wheels.contains(_ex)) {
            return;
        }
        wheels.remove(_ex);
        _ex.removeFromMycar(this);
    }

    /**
     * accessor for association to spare
     * reference to the spared wheel
     */
    public Wheel getSpare() {
        return spare;
    }

    /**
     * accessor for association to spare
     * @see {@link getSpare}
     */
    public void insertInSpare(Wheel _in) {
        if (spare == _in) {
            return;
        }
        spare = _in;
    }

    /**
     * accessor for association to spare
     * @see {@link getSpare}
     */
    public void removeFromSpare(Wheel _ex) {
        if (spare != _ex) {
            return;
        }
        spare = null;
    }

    /**
     * accessor for association to license
     */
    public java.util.Set<License> getLicense() {
        java.util.HashSet<License> _result = new java.util.HashSet<License>();
        for (Owner _assoc : license) {
            _result.add(_assoc.getLicense());
        }
        return _result;
    }

    /**
      * accessor to the association class objects
      */
    public java.util.Set<Owner> getLicenseOwner() {
        return java.util.Collections.unmodifiableSet(license);
    }

    /**
     * accessor for association to license
     */

    /**
      * insert into association, generate linked association class object
      * @param _other partner object
      * @return newly generated linked association class object
      */
    public Owner insertInLicense(License _other) {
        Owner _assoc = new Owner(_other, this);
        license.add(_assoc);
        _other.insertInCar(_assoc);
        return _assoc;
    }

    /**
      * insert linked association class object into association
      * @param _other partner object
      * @return newly generated linked association class object
      */
    public void insertInLicense(Owner _assoc) {
        if (license.contains(_assoc)) {
            return;
        }
        if (_assoc.getCar() != this) {
            throw new RuntimeException("inconsistent association object");
        }
        license.add(_assoc);
    }

    /**
     * accessor for association to license
     */

    /**
      * remove entry from association indexed by its association class object
      */
    public void removeFromLicense(Owner _assoc) {
        if (!license.contains(_assoc)) {
            return;
        }
        license.remove(_assoc);
        _assoc.getLicense().removeFromCar(_assoc);
    }

    /**
     * accessor for association to licensePlate
     */
    public LicensePlate getLicensePlate() {
        return licensePlate;
    }

    /**
     * accessor for association to licensePlate
     */
    public void insertInLicensePlate(LicensePlate _in) {
        if (licensePlate == _in) {
            return;
        }
        if (licensePlate != null) {
            licensePlate.removeFromCar(this);
        }
        licensePlate = _in;
        _in.insertInCar(this);
    }

    /**
     * accessor for association to licensePlate
     */
    public void removeFromLicensePlate(LicensePlate _ex) {
        if (licensePlate != _ex) {
            return;
        }
        licensePlate = null;
        _ex.removeFromCar(this);
    }

    /**
     * accessor for association to engine
     */
    public Engine getEngine() {
        return engine;
    }

    /**
     * accessor for association to engine
     */
    public void insertInEngine(Engine _in) {
        if (engine == _in) {
            return;
        }
        if (engine != null) {
            engine.removeFromCar(this);
        }
        engine = _in;
        _in.insertInCar(this);
    }

    /**
     * accessor for association to engine
     */
    public void removeFromEngine(Engine _ex) {
        if (engine != _ex) {
            return;
        }
        engine = null;
        _ex.removeFromCar(this);
    }

    // ////////////////////////////////////////////////////////////////////////
    // generated abstract method declaration
    // ////////////////////////////////////////////////////////////////////////

    // ////////////////////////////////////////////////////////////////////////
    // generated normal method implementations
    // ////////////////////////////////////////////////////////////////////////

    /**
     * define if the car is driving
     * @param  speed
     * @return
     * @throws  WhatAMessException
     * @throws  NotReallySeriousException  this is a comment; this is the second comment
     */
    public final boolean drive(int speed)
        throws WhatAMessException, NotReallySeriousException {
        Boolean returnValue = null;

        /* <!-- PROTECTED REGION ID(java.moperation.implementation.operation.code._11_5_6340215_1179222914421_654419_131) ENABLED START --> */
        /* <!-- TODO put your own implementation code here --> */

        /* <!-- PROTECTED REGION END --> */
        return returnValue;
    }

    // ////////////////////////////////////////////////////////////////////////
    // generated getter and setter method implementations
    // ////////////////////////////////////////////////////////////////////////

    /**
      * describs the maximum speed of a car
      */
    public int getMaxSpeed() {
        return maxSpeed;
    }

    /** @see {@link getMaxSpeed} */
    public void setMaxSpeed(int _maxSpeed) {
        maxSpeed = _maxSpeed;
    }

    // ////////////////////////////////////////////////////////////////////////
    // own implementations
    // ////////////////////////////////////////////////////////////////////////

    /* <!-- PROTECTED REGION ID(java.mclassifier.implementation.owncode.code._11_5_6340215_1177945913718_872802_139) ENABLED START --> */
    /* <!-- TODO put your own implementation code here --> */

    /* <!-- PROTECTED REGION END --> */
}
